src/aks-preview/azext_aks_preview/tests/latest/test_decorator.py

```diff
@@ -1481,6 +1521,64 @@ def test_set_up_network_profile(self):
         )
         self.assertEqual(dec_mc_2, ground_truth_mc_2)
 
+        # dual-stack
+        dec_3 = AKSPreviewCreateDecorator(
+            self.cmd,
+            self.client,
+            {
+                "load_balancer_sku": None,
+                "load_balancer_managed_outbound_ip_count": None,
+                "load_balancer_managed_outbound_ipv6_count": 3,
+                "load_balancer_outbound_ips": None,
+                "load_balancer_outbound_ip_prefixes": None,
+                "load_balancer_outbound_ports": None,
+                "load_balancer_idle_timeout": None,
+                "outbound_type": None,
+                "network_plugin": "kubenet",
+                "pod_cidr": None,
+                "service_cidr": None,
+                "pod_cidrs": "10.246.0.0/16,2001:abcd::/64",
+                "service_cidrs": "10.0.0.0/16,2001:ffff::/108",
+                "ip_families": "IPv4,IPv6",
+                "dns_service_ip": None,
+                "docker_bridge_cidr": None,
+                "network_policy": None,
+            },
+            CUSTOM_MGMT_AKS_PREVIEW,
+        )
+        mc_3 = self.models.ManagedCluster(location="test_location")
+        dec_mc_3 = dec_3.set_up_network_profile(mc_3)
+
+        network_profile_3 = self.models.ContainerServiceNetworkProfile(
+            network_plugin="kubenet",
+            pod_cidr="10.246.0.0/16",
+            service_cidr="10.0.0.0/16",
+            dns_service_ip=None,  # overwritten to None
+            docker_bridge_cidr=None,  # overwritten to None
```

Comment:
According to the implementation (base function) in [azure-cli](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/decorator.py#L4731-L4755), we would overwrite `dns_service_ip/docker_bridge_cidr` to `None` when any of `network_plugin/pod_cidr/service_cidr/dns_service_ip/docker_bridge_address/network_policy` is specified but they themselves are not. Take an example, if user only specifies the new parameters like `pod_cidrs`, `service_cidrs` or `ip_families`, we would still take the default values provided by SDK for `dns_service_ip` and `docker_bridge_cidr`.

Reply:
updated to capture this functionality. if ip_families is provided then defaults should be kept (if other things are provided)